lib/srv/db/sqlserver/protocol/packet.go

```diff
@@ -139,7 +140,7 @@ func NewBasicPacket(header PacketHeader, data []byte) (*BasicPacket, error) {
 func ToSQLPacket(p *BasicPacket) (out Packet, err error) {
 	defer func() {
 		if r := recover(); r != nil {
-			err = trace.BadParameter("failed to convert packet to SQL packet: %v", r)
+			err = trace.BadParameter("failed to convert packet to SQL packet: %v @ %v", r, debug.Stack())
```

Comment:
do we know how often this happens? this is probably spamming the logs already because of the old driver.

cc @gabrielcorado

Reply:
The spam is possible, although it would manifest as a regular error so I'm not sure how significant would that be. I'll make the other one warning as well.

Reply:
If anyone is using a newer version of the TDS protocol (which does not officially support it), this could be throwing a lot of errors. Other than that, although we don't block the connection due to this, it is very confusing to see a lot of malformed audit events and have no clue about what is happening.

With that said, I still think this should be considered an error, as users are not receiving the full audit logging for SQL Server when we receive an unrecognizable packet. (Could this be considered a security issue if users are trying to hide a malicious command by using packets we cannot parse?)

Reply:
Looking at usage of `ErrorContext()` we typically reserve it for fatal, instance-wide errors affecting the entirety of the service rather than a single user session, so a warning seems the correct level in this case.